cmd/controller/main.go

```diff
@@ -79,6 +80,16 @@ func main() {
 	if err != nil {
 		logger.Fatalf("Error building kubernetes clientset: %v", err)
 	}
+	// Knative requires a Kubernetes cluster v1.10 or newer, making debugging hard. This makes it easier to
+	// figure out if kubernetes meet the requirement.
+	v, err := kubeClient.Discovery().ServerVersion()
```

Comment:
This seems like a useful thing to add to knative/pkg, so all the Knative components can reuse it.

Reply:
Looks already pretty well factored, but other things could use this for sure.